### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by vkarpov15**
Borrowed heavily from https://github.com/stargate/stargate-mongoose/blob/cbd25c0949616520920e1e1947181d8ab5b9fa90/.github/workflows/ci-tests.yml#L18-L29, thanks @kathirsvn . This Workflow clones the stargate-mongoose repo and uses the `bin/start_json_api.sh` script to start stargate-mongoose.

@kathirsvn @jeffreyscarpenter what do you think about including the `bin/start_json_api.sh` and `bin/docker-compose.yml` files in the npm package? Given that stargate's JSON API is relatively new and there isn't much content out there on how to run it, I think it would be nice to include a nice one-line command to run the JSON API.

### CODE REVIEW

**Review by kathirsvn — APPROVED** [review]
LGTM

**Comment by kathirsvn** [comment]
@vkarpov15 I like the idea of packaging the scripts and publishing it to npm as part of stargate-mongoose. I believe the idea behind that is that, with some `npm run` command developers should be able to start the JSON API on local? If not, could you please explain how will the developer experience be like?

**Comment by vkarpov15** [comment]
An `npm run` command would be great, but that might be a bit tricky because I'm not sure you can use a `.sh` file for a Node command line script (this blog is a good write up on how to do that: https://medium.com/netscape/a-guide-to-create-a-nodejs-command-line-package-c2166ad0452e). I think allowing running `./node_modules/stargate-mongoose/bin/start_json_api.sh` is a decent start to allow running the JSON API without cloning a separate GitHub repo.

**Comment by kathirsvn** [comment]
Got it @vkarpov15. I have created an issue for the same.
https://github.com/stargate/stargate-mongoose/issues/49

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,40 +13,33 @@ jobs:
...
-      - run: npm run build
-        working-directory: stargate-mongoose
-      - run: node ../.scripts/create-tarball.js
+      - name: Wait for services
+        run: |
+          while ! nc -z localhost 8080; do sleep 1; done
+          while ! nc -z localhost 8081; do sleep 1; done
         working-directory: stargate-mongoose
+      ## Set up and run ecommerce sample app tests
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
```
[the remark this answers is not in the document]
Comment (vkarpov15) [reply .github/workflows/test.yml:37]: @jeffreyscarpenter security best practices for GitHub actions: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions . That hash is equivalent to `actions/checkout@3.0.2`, but also protects against someone deleting or changing the `3.0.2` tag.

**Comment by jeffreyscarpenter** [comment]
@vkarpov15 try again to see if you can merge?

**Pull request merged**